draft-tus-resumable-uploads-protocol.md

```diff
@@ -421,3 +421,10 @@ Following **approaches** have already been considered in the past. All except th
 - Use a new response header `Resumable-Uploads: https://example.org/files/*` to indicate under which endpoints support for resumable uploads is available.
 
 **Send a 104 intermediate response to indicate support.** The clients normally starts a traditional upload and includes a header indicate that it supports resumable uploads (e.g. `Upload-Offset: 0`). If the server also supports resumable uploads, it will immediately respond with a 104 intermediate response to indicate its support, before further processing the request. This way the client is informed during the upload whether it can resume from possible connection errors or not. While an additional roundtrip is avoided, the problem with that solution is that many HTTP server libraries do not support sending custom 1XX responses and that some proxies may not be able to handle new 1XX status codes correctly.
+
+**Send a 103 Early Hint response to indicate support.** This approach is the similar to the above one with one exception: Instead of a new `104 (Upload Resumption Supported)` status code, the existing `103 (Early Hint)` status code is used in the intermediate response. The 103 code would then be accompanied by a header indicating support for resumable uploads (e.g. `Resumable-Uploads: 1`). It is unclear whether the Early Hints code is appropriate for that as it is currently only used to indicate resources for prefetching them.
```

Comment:
The 103 spec says 

>    The 103 (Early Hints) informational status code indicates to the
   client that the server is likely to send a final response with the
   header fields included in the informational response.

So in theory, this is an early hint about the fields that would appear in the final response. In theory it should work for any type of header field. In theory, the nice thing with early hints is that if an intermediary drops the 103 (which is allowed) then the client still expects that the 200 response is the final answer about if the hint was true or not. 104 doesn't seem to accommodate that (yet) and probably needs some more thought.

In practice, I agree with this statement that is is unclear if 103 implementations really support the spirit of the spec, or anything beyond Link headers

Reply:
Makes sense. It's too late to do feature detection at the point where the response is received, and we are basically using the presence of 104 to signal support, so it might not be the best fit

Reply:
Good insights, thank you! Once we are more certain whether to go with 103, 104 or none of these, we can expand the details.